src/stores/PeopleContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { Context } from '../types/Context';
+import { Filter } from '../types/Filter';
+
+const initState: Context = {
+  people: [] as Person[],
+  isLoading: false,
+  loadingError: false,
+  filterPeople: () => [],
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const PeopleContext = React.createContext(initState);
+
+export const PeopleProvider: React.FC<Props> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingError, setLoadingError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setLoadingError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const filterPeople = ({
```

Comment:
For me, it's a bit inconsistent

Chose one:
- move this function into a separate file (utils.ts or smth like that) pass everything (including the people array) as an argument and use this function, import it in the `PeopleTable` component
- get all params from `useSearchParams` hook here calculate an array of filtered people and pass the prepared array as a context value